crates/red_knot_python_semantic/src/module_resolver/resolver.rs

```diff
@@ -627,6 +692,21 @@ impl PackageKind {
     }
 }
 
+pub(super) struct ResolverContext<'db> {
+    pub(crate) db: &'db dyn Db,
+    pub(crate) target_version: PythonVersion,
+}
+
+impl<'db> ResolverContext<'db> {
+    pub(super) fn new(db: &'db dyn Db, target_version: PythonVersion) -> Self {
+        Self { db, target_version }
+    }
+
+    pub(super) fn vendored(&self) -> &VendoredFileSystem {
+        self.db.vendored()
+    }
+}
```

Comment:
The reason I put this in its own module (tiny though it is) was because it felt strange to me to import it from `resolver.rs` in `path.rs`, considering that `path.rs` is a "lower-level" module than `resolver.rs`. But maybe that's just me "thinking in Python" a bit too much, where it's really important to avoid cyclic dependencies between modules :-)

Reply:
I think this is how it is intended in Rust where lower level modules have access to the parent but the inner modules are private to the parent.

Reply:
Ah I guess the real solution here would be to make `path` a submodule of `resolver`